plugin/src/main/java/at/helpch/chatchat/listener/ChatListener.java

```diff
@@ -31,33 +31,30 @@ public void onChat(final AsyncPlayerChatEvent event) {
         final var user = plugin.usersHolder().getUser(player);
 
         final var format = FormatUtils.findFormat(player, plugin.configManager().formats());
-        final var oldChannel = user.channel(); // save their old channel in case they use a message prefix
 
         final var channelByPrefix =
                 ChannelUtils.findChannelByPrefix(List.copyOf(plugin.configManager().channels().channels().values()), event.getMessage());
-        channelByPrefix.ifPresent(channel -> {
-            if (user.canUse(channel)) user.channel(channel);  // set the channel if their message starts with a channel prefix & has perms
-        });
 
         final var message = channelByPrefix.isEmpty()
                 ? event.getMessage()
                 : event.getMessage().replaceFirst(Pattern.quote(channelByPrefix.get().messagePrefix()), "");
 
-        final var newChannel = user.channel();
+        final var channel = user.channel();
```

Comment:
Is this really needed? Maybe instead it should be something like this:
`final var channel = channelByPrefix.isPresent() ? channelByPrefix.get() : user.channel();` or maybe don't put it in a variable at all.

Reply:
Other than this it all seems pretty good. and yes I am nitpick rn lol